### ISSUE TITLE
Better handling of unsupported region.

### ISSUE THREAD

**Issue by tam203** [informatics-lab/aml-jupyterhub#33]
![image](https://user-images.githubusercontent.com/1567535/96745926-8ae66100-13be-11eb-9d77-a0bcd4a4a399.png)

![image](https://user-images.githubusercontent.com/1567535/96745949-920d6f00-13be-11eb-8a78-669d880c7f71.png)

```
[E 2020-10-21 15:55:45.609 JupyterHub pages:247] Failed to spawn single-user server with form
    Traceback (most recent call last):
      File "/home/azureuser/miniconda3/envs/aml/lib/python3.8/site-packages/jupyterhub/handlers/pages.py", line 244, in post
        options = await maybe_future(spawner.options_from_form(form_options))
      File "/home/azureuser/aml-jupyterhub/aml_jupyterhub/aml_spawner.py", line 150, in options_from_form
        self.vm_size = VM_SIZES[self.location][size_selected]
    KeyError: 'westeurope'
```

Minor bug, just making a note since I came across it. We can get to this as and when.

**Comment by nbarlowATI** [comment]
I think a better way of dealing with making sure the selected VM size is available in the desired region could be something like:
 
Have a dictionary containing an preference-ordered list of acceptable sizes for each of "Small", "Medium", ...
```
VM_SIZES = {
  "Small" : ["Standard_NC1", "some-other-small-VM", ...],
  "Medium":["Standard_NC3", "some-other-medium-VM", ...]
  ...
}
```
We then use the Azure python SDK to get a list of available VM sizes in the region (using the "location" env var), and go through those lists, picking the first entry if it is available, or moving on to the second if it isn't.   If none of the "Small" VMs we have specified are available, then the user doesn't get "Small" as an option on the form. 
This way we can add new types of VM to the front of the list if Azure rolls them out, but we will still have the existing types to fall back on if the new ones aren't available in our region.

This is implemented in #38 and seems to work OK, though i might test further  by filling one of the dictionary items with non-existent VM sizes.

**Comment by tam203** [comment]
Neat solution! 👏
